pkg/build/builder/docker.go

```diff
@@ -69,10 +70,20 @@ func (d *DockerBuilder) Build() error {
 	}
 	sourceInfo, err := fetchSource(d.dockerClient, buildDir, d.build, d.urlTimeout, os.Stdin, d.gitClient)
 	if err != nil {
+		d.build.Status.Reason = api.StatusReasonFetchSourceFailed
+		d.build.Status.Message = StatusMessageFetchSourceFailed
+		if updateErr := updateStatusRetrier(d.build, d.client); updateErr != nil {
+			glog.V(0).Infof("error: An error occured while updating the build status: %v", updateErr)
+		}
 		return err
 	}
+
 	if sourceInfo != nil {
-		updateBuildRevision(d.client, d.build, sourceInfo)
+		glog.V(4).Infof("Setting build revision with details %#v", sourceInfo)
+		updateBuildRevision(d.build, sourceInfo)
+		if updateErr := updateStatusRetrier(d.build, d.client); updateErr != nil {
```

Comment:
am I missing something here?  updateBuildRevision just updates the local d.build object, and updateStatusRetrier grabs the remote build object and sets message/reason on it and posts it back, but it does not set the revision.

and if i'm right and if we don't have a test case that this tripped, that's bad.